### ISSUE TITLE
Conditions not being refreshed correctly on Custom System Builder

### ISSUE THREAD

**Issue by Kiiinji** [mclemente/fvtt-condition-lab-triggler#44]
**Describe the bug**
Label informations from the Custom System Builder are not read correctly by the Condition trigglers when the character sheet is not opened.

**To Reproduce**
- Create a label with an automatically calculated variable, like for example a percentage of your player's stamina.
- This label will now automatically refresh when your stamina is reduced or highered.
- The condition trigglers will read the value of the label correctly when the sheet is opened
- But if you close the character sheet and reduce or higher the stamina inside the actor's token, the effect trigger won't read anything
- However, when opening the sheet again, it will now refresh and update

**Expected behavior**
The effect should be triggered even when the Sheet is closed (the value updates even when the sheet is closed, I verified it with an attribute bar reading the value of the stamina_percentage)

**Screenshots**
[Here is a video](https://gyazo.com/401d826268facef10bf2475bd0e1a2d6)
(as you can see in the gif, the value updates to "2", so the effect should triggered, but it doesn't until I open the Sheet.

Also, here is my trigger value
![image](https://github.com/mclemente/fvtt-condition-lab-triggler/assets/39131872/26292c84-9bbe-41f3-b4e3-925045d85a94)

**Environment Info (please complete the following information):**
 - Foundry Version: 11.315
 - Game System / Version: Custom System Builder / 3.0.0
 - CUB Version 1.5.5
 - Browser : Official app
 - OS : Windows 10

**Comment by Kiiinji** [comment]
Still ongoing :(

**Comment by mclemente** [comment]
Could you share the template you're using? I'm not used to CSB's syntax.

And I think that's an issue on the system's side. The system is doing some update when the sheet is being rendered instead of updating when the actor's data is updated.

**Comment by Kiiinji** [comment]
Many other modules update instantly weither the sheet is opened or not, and values like the hunger or stamina meters are updated through the sheet and the token even when the sheet is closed.

I escalated this issue to the CSB author and he mentioned that this issue was on Condition lab's side.

As for the template, what exatcly do you need me to share?

**Comment by mclemente** [comment]
> As for the template, what exactly do you need me to share?

That actor's template, _Jouertemp. And which fields do I need to fill in to get that ratio?

**Comment by Kiiinji** [comment]
Where can I find this file?

**Comment by mclemente** [comment]
Right-click the `_Jouertemp` temp actor on the Actors Sidebar and choose to Export Data, then upload it here

**Comment by Kiiinji** [comment]
For the field, it's the one at the bottom left, the first "hidden" data
![image](https://github.com/user-attachments/assets/4c5e9509-349b-41a1-951d-1df2c7dd06fe)


[fvtt-Actor-_joueurtemp-5AYyd3m5x11651UC.json](https://github.com/user-attachments/files/16878858/fvtt-Actor-_joueurtemp-5AYyd3m5x11651UC.json)

**Comment by mclemente** [comment]
The system *is* updating during render [here](https://gitlab.com/custom-system-builder/custom-system-builder/-/blob/7357c2e03bc0ee8d9439dcd21bfa3ecb25c01560/src/module/sheets/components/DynamicTable.ts#L468) instead of doing it on `Actor#_onUpdate`.

You might be able to circumvent this issue by using an advanced trigger that calculates the percentage.

Actor Update Property: `system.props.STAMINA`
Actor Comparison Property:`system.props.STAMINA_MAX`
Operator: `>` (or `≥`, not sure which you want)
Value: `50%`

Repeat for the opposite, with a `<` or `≤` Operator.

---

Thank you for providing the template.

While investigating this issue I've found out another issue with the way the trigger checks equality/inequality, so avoid the `=` and `≠` for percentages.

**Comment by Kiiinji** [comment]
This solution doesn't work.

I need the status to show below a threshold and hide when under or over this threshold.

This means that I need two triggers for the status to be removed and not just one.

For example:
```

If stamina < 75% & > 50% {
status 1
} else if stamina < 50 % & > 30% {
status 2
}
```
etc etc..

If I set the remove trigger for "status 1" to "> 50%" then it won't disappear if my stamina is over 75%, which is an issue.

**Comment by Kiiinji** [comment]
up :(

**Comment by mclemente** [comment]
Sorry, I've been a bit busy :(

Hm, try `< 75%` for the remove trigger `> 50%` for the apply trigger, then, on the "status 2", add a macro like this:

```js
const status1 = CONFIG.statusEffects.find(e => e.name === NAME_OF_STATUS_1);
token.toggleEffect(status1, { active: false });
```

This will probably be triggered on every update, though, so it might be annoying.

I'm really sorry, but I don't think I can fix this, at least not for V11, my time for projects has become limited and this project is the toughest to keep up to date.

**Comment by Kiiinji** [comment]
Hello

Thank you for this reply!

Sadly, this is not a solution viable for me, I guess I'll have to keep all character sheets opened for now.

Thank you for your support tho.

Best,

**Comment by Kiiinji** [comment]
Hi, since everything as been updated to v12, are there any news or hope for this issue?
